src/gwt/src/org/rstudio/studio/client/workbench/views/source/SourceColumn.java

```diff
@@ -0,0 +1,1215 @@
+      commands_ = commands;
+      binder.bind(commands_, this);
+
+      events_ = events;
+      editingTargetSource_ = editingTargetSource;
+      fileContext_ = fileContext;
+      server_ = sourceServerOperations;
+
+
+      events_.addHandler(FileTypeChangedEvent.TYPE, event -> manageCommands(false));
+      events_.addHandler(SourceOnSaveChangedEvent.TYPE, event -> manageSaveCommands());
+      events_.addHandler(SynctexStatusChangedEvent.TYPE, event -> manageSynctexCommands());
+
+      initialized_ = true;
+   }
+
+   public void loadDisplay(String name,
+                           Source.Display display,
+                           SourceColumnManager manager)
+   {
+      name_ = name;
+      display_ = display;
+      manager_ = manager;
+
+      display_.addSelectionHandler(this);
+      display_.addTabClosingHandler(this);
+      display_.addTabCloseHandler(this);
+      display_.addTabClosedHandler(this);
+      display_.addTabReorderHandler(this);
+
+      ensureVisible(false);
+   }
+
+   public String getName()
+   {
+      return name_;
+   }
+
+   public EditingTarget getActiveEditor()
+   {
+      return activeEditor_;
+   }
+
+   public ArrayList<EditingTarget> getEditors()
+   {
+      return editors_;
+   }
+
+   public ArrayList<EditingTarget> getDirtyEditors(final EditingTarget excludeEditor)
+   {
+      ArrayList<EditingTarget> dirtyEditors = new ArrayList<>();
+      for (EditingTarget target : editors_)
+      {
+         if (excludeEditor != null && target == excludeEditor)
+            continue;
+         if (target.dirtyState().getValue())
+            dirtyEditors.add(target);
+      }
+      return dirtyEditors;
+   }
+
+   public ArrayList<EditingTarget> getUnsavedEditors(int type, Set<String> ids)
+   {
+      ArrayList<EditingTarget> unsavedEditors = new ArrayList<>();
+      for (EditingTarget target : editors_)
+      {
+         if (!isUnsavedTarget(target, type))
+            continue;
+         if (ids != null && !ids.contains(target.getId()))
+            continue;
+         unsavedEditors.add(target);
+      }
+      return unsavedEditors;
+   }
+
+   public ArrayList<Integer> getTabOrder()
+   {
+      return tabOrder_;
+   }
+
+   public Widget asWidget()
+   {
+      return display_.asWidget();
+   }
+
+   // Display wrapper methods
+   public void cancelTabDrag()
+   {
+      display_.cancelTabDrag();
+   }
+   
+   public void closeTab(Widget child, boolean interactive)
+   {
+      display_.closeTab(child, interactive);
+   }
+
+   public void closeTab(Widget child, boolean interactive, Command onClosed)
+   {
+      display_.closeTab(child, interactive, onClosed);
+   }
+   
+   public void closeTab(int index, boolean interactive)
+   {
+      display_.closeTab(index, interactive);
+   }
+
+   public void closeTab(int index, boolean interactive, Command onClosed)
+   {
+	   display_.closeTab(index, interactive, onClosed);
+   }
+
+   public void closeAllLocalSourceDocs()
+   {
+      // TODO
```

Comment:
Is this intentionally left as "TODO"?

Reply:
No, I remove this and added an option to `SouceColumnManager's` `closeAllLocalSourceDocs` function to accept a single column.